state/immutable-trie/storage.go

```diff
@@ -17,81 +16,39 @@ var (
 	codePrefix = []byte("code")
 )
 
-type Batch interface {
-	Put(k, v []byte)
-	Write()
-}
-
 // Storage stores the trie
 type Storage interface {
-	Put(k, v []byte)
-	Get(k []byte) ([]byte, bool)
-	Batch() Batch
+	kvdb.KVBatchStorage
```

Comment:
How about keeping those functions? It was a 'duck type' contraction. 

As a matter of fact, high-level modules use abstractions directly from low-level modules, this approach violates the principle of dependency inversion (DIP).

I would suggest that we keep those interface, which implementation could be replaced at any moment.